browser/brave_tab_helpers.cc

```diff
@@ -67,6 +78,21 @@ void AttachTabHelpers(content::WebContents* web_contents) {
   BraveDrmTabHelper::CreateForWebContents(web_contents);
 #endif
 
+#if BUILDFLAG(ENABLE_BRAVE_WAYBACK_MACHINE)
+  auto* context = web_contents->GetBrowserContext();
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kDisableBraveWaybackMachineExtension) &&
+      !brave::IsTorProfile(context)) {
+    BraveWaybackMachineTabHelper::CreateForWebContents(web_contents);
+    auto* tab_helper =
+        BraveWaybackMachineTabHelper::FromWebContents(web_contents);
+    tab_helper->SetInfoBarManager(
```

Comment:
I think the InfoBar stuff should be moved to the delegate so we don't have to make multiple method calls. Also the convention for simple settings is `set_delegate`

Reply:
Done. `BraveWaybackMachineDelegateImpl::AttachTabHelperIfNeeded()` is created for this.